server/client.go

```diff
@@ -3238,7 +3238,7 @@ func (c *client) processInboundClientMsg(msg []byte) bool {
 		if c.opts.NoResponders {
 			if sub := c.subForReply(c.pa.reply); sub != nil {
 				proto := fmt.Sprintf("HMSG %s %s 16 16\r\nNATS/1.0 503\r\n\r\n\r\n", c.pa.reply, sub.sid)
-				c.sendProtoNow([]byte(proto))
```

Comment:
Should this just be `queueOutbound()` and set pcd entry since we are in the readloop?

Reply:
I thought about it since the subscription is from its own client, that would work. Ok, will update this.

Reply:
Could be either way for this one I suppose.

Reply:
Went the pcd route and took that opportunity to make that a function since we do that in 3 places now, and there was a bug with sendOK() that was putting in the map but not bumping fsp (when adding for the first time), which would have created a negative fsp at the time of the flushing.

Reply:
Thinking of it, the sendOK() should really just be a enqueueProto() since there is no concept of client to flush other than itself, which the writeLoop would take care of.